File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/EventHall.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class EventHall {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    private RectangleGeometry geometry;

    @OneToMany(mappedBy = "eventHall", fetch = FetchType.EAGER)
    private Set<Layout> layouts = new HashSet<>();

    @ManyToOne(fetch = FetchType.EAGER)
    private Location location;
}

File: frontend/src/app/dtos/event-hall.ts
import {CreateLayout} from './layout';
import {Location} from './location';
import {RectangleGeometry} from './rectangle-geometry';

export class EventHall {
  constructor(
    public id: number,
    public name: string,
    public geometry: RectangleGeometry,
    public location: Location
  ) {
  }
}

export class CreateEventHall {
  id: number;
  name: string;
  location: Location;
  x: number;
  y: number;
  width: number;
  height: number;
  layout: CreateLayout;
}

export class EventHallOverview {
  id: number;
  name: string;
  location: Location;
}

export class TemporaryEventHallInformation {
  name: string;
  location: Location;
}



